packages/embark-specialconfigs/src/index.js

```diff
@@ -177,6 +178,22 @@ class SpecialConfigs {
     });
   }
 
+  registerAddressHandlerAction() {
+    this.embark.registerActionForEvent('contract:address:handler', async (params, cb) => {
+      const contract = params.contract;
+      try {
+        const dependencies = await this.getOnDeployLifecycleHookDependencies({
+          contractConfig: contract,
+          logPrefix: `${contract.className} > addressHandler >`
+        });
+        const address = await contract.addressHandler(dependencies);
```

Comment:
Instead of allowing `contract.address` to be a function, I've introduced a new property `addressHandler`. This is because:

1. I think it makes it a bit more clear what's going on, you either have a static address, or you need to compute it
2. Changing address to be a function as well possibly breaks a lot of places in Embark unless they are fully checked, which is kinda hard without a proper test suite. In fact, I immediately ran into runtime errors when trying to pass `address` as a function.

Lemme know if you have strong feelings about not introducing a new property.

Reply:
At first I was sad it was not called `address`, but you are right, it's clearer for the user with `addressHandler` and it also doesn't break Embark like I expect it would.

Reply:
Personally, I would prefer that it stays as `address` for consistency's sake so that we don't keep adding new (and different/random/inconsistent) property names for functions later on down the track. If enough people agree with that, maybe this could be handled in a separate PR.

Reply:
Very fair. It just means we need to be more careful and test thoroughly.